### PR TITLE
Add test for trailing space in location in hive metadata

### PR DESCRIPTION
**Pull request by pajaks**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by pajaks** [comment]
@ebyhr @findepi Could you run with secretes?

**Comment by ebyhr** [comment]
/test-with-secrets sha=3527cc78f165fd241dedb3dc63f46dd8281d6ea6

**Comment by pajaks** [comment]
Rebased to include fix for skipped tests in CI

**Comment by ebyhr** [comment]
/test-with-secrets sha=2b64fc19464e0ea355d949ea3396c550aefd58b4

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 1]
```diff
@@ -2595,6 +2598,61 @@ public void testTableCreation()
         }
     }
 
+    @Test
+    public void testTableCreationWithTrailingSpaceInLocation()
+            throws Exception
+    {
+        SchemaTableName temporaryTable = temporaryTable("test_table_creation_with_trailing_space_in_location_" + randomNameSuffix());
```
Comment (findinpath) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:2605]: field name `temporaryTable` -> `tableName`

**Review by findinpath — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 2]
```diff
@@ -5434,6 +5492,15 @@ protected static Map<String, Object> createTableProperties(HiveStorageFormat sto
         return createTableProperties(storageFormat, ImmutableList.of());
     }
 
+    protected static Map<String, Object> createTableProperties(HiveStorageFormat storageFormat, String externalLocation)
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:5495]: nit: We can change to `private`.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Could you confirm CI failure? 
```
Error:  Tests run: 95, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 163.58 s <<< FAILURE! - in io.trino.plugin.hive.TestHive
Error:  io.trino.plugin.hive.TestHive.testTableCreationWithTrailingSpaceInLocation  Time elapsed: 0.017 s  <<< FAILURE!
java.lang.IllegalArgumentException: Wrong FS: hdfs://hadoop-master:9000/user/hive/warehouse/tmp_trino_test_test_table_creation_with_trailing_space_in_location_5w96076bmm_754e382b8d0c4e1793f934108d5871d8 , expected: file:///
	at org.apache.hadoop.fs.FileSystem.checkPath(FileSystem.java:737)
	at org.apache.hadoop.fs.RawLocalFileSystem.pathToFile(RawLocalFileSystem.java:87)
	at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:661)
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileLinkStatusInternal(RawLocalFileSystem.java:987)
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:656)
	at org.apache.hadoop.fs.ChecksumFileSystem.delete(ChecksumFileSystem.java:642)
	at org.apache.hadoop.fs.FilterFileSystem.delete(FilterFileSystem.java:264)
	at io.trino.plugin.hive.AbstractTestHive.testTableCreationWithTrailingSpaceInLocation(AbstractTestHive.java:2658)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by pajaks** [comment]
Problem is caused by my changes. I am looking into it.

**Comment by pajaks** [comment]
Test should work now. @ebyhr Can you run with secretes?

**Comment by ebyhr** [comment]
/test-with-secrets sha=657e0685270c67052aa8609b3f93405c50470daa

**Comment by ebyhr** [comment]
The above failure is unrelated to this change.

**Pull request merged**